### DISCUSSION TITLE
Multiple Beamlets.

### DISCUSSION THREAD

**Discussion by XimmiChangeXi** [AOtools/aotools#87] [Q&A]
Greetings everyone, I am currently using aotools for atmospheric propagation. I have successfully simulated a single beam in the atmosphere at certain target. However, I wanted to know that is there anyway I can simulate multiple beams at once. Does aotools support this.? To elaborate further, I have simulated at source (z=0) three gaussian beams, and when I try to pass them to the propagating function (I am using angularSpectrum method), I have to pass each beam separately. After that I get same scenario at z=0. I wanted them to converge to the point. Does anyone know how can I do that? all beams should converge at that point. 
(The plots I gained are also attached).. 
![output_beams_at_z_5000](https://user-images.githubusercontent.com/108569019/189291095-bf6b2cbe-2bbb-4f77-9817-e03a1f7a4378.png)
![3_beams_at_z_0](https://user-images.githubusercontent.com/108569019/189291103-cc582f4d-4784-4eaf-948b-5d1fcc37d516.png)

**Comment by matthewtownson** [comment]
This is something I think you could do once rather than separately. By getting the beams to converge do you mean you will angle the source so the beams overlap at a specified z distance? If so, then I think you could add a gradient to the phase around the edge sources, this should then move them laterally as the beams propagate. This is something you would have to add into the phase yourself as AOtools does not implement this.

**Reply by XimmiChangeXi** [reply]
Yeah, my goal is to converge the beams on to a single point by angling them. But, pardon me, I don't understand about "Adding gradient to the phase around the edge sources." Could please elaborate this?

**Reply by matthewtownson** [reply]
If you propagate the middle Gaussian then it will continue to remain in the centre of the field. If you take the left Gaussian then you want to add a divert its path to the middle. The angle you want to tilt the beam by will be related to the propagation distance (the shorter the propagation distance, the larger the angle), To divert a beam we can modify the phase part of a complex amplitude adding a tilt. So, you can draw a box around the outer edge Gaussians and apply a tilt to the phase of the wavefront to divert the beam.

Alternatively you could add a focus to the whole phase across the field (equivalent to adding a lens). Again, to work out the amount of focus you would require depends on the propagation distance you want to have the beams overlap.
